core/templates/services/playthrough.service.spec.ts

```diff
@@ -16,246 +16,287 @@
+
+describe('PlaythroughService', () => {
+  let playthroughService: PlaythroughService = null;
+  let httpTestingController: HttpTestingController = null;
+  let learnerActionObjectFactory: LearnerActionObjectFactory = null;
+  let explorationFeaturesService: ExplorationFeaturesService = null;
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+
+    httpTestingController = TestBed.get(HttpTestingController);
+    playthroughService = TestBed.get(PlaythroughService);
+    learnerActionObjectFactory =
+      TestBed.get(LearnerActionObjectFactory);
+    explorationFeaturesService =
+      TestBed.get(ExplorationFeaturesService);
+  });
+
+  afterEach(()=> {
+    httpTestingController.verify();
+  });
+
+  describe('.initSession()', () => {
+    it('stores the correct values', () => {
       this.expId = 'expId';
       this.expVersion = 1;
       this.playthroughRecordingProbability = 1.0;
 
-      this.PlaythroughService.initSession(
+      playthroughService.initSession(
         this.expId, this.expVersion, this.playthroughRecordingProbability);
 
-      var playthrough = this.PlaythroughService.getPlaythrough();
+      var playthrough = playthroughService.getPlaythrough();
       expect(playthrough.expId).toEqual(this.expId);
       expect(playthrough.expVersion).toEqual(this.expVersion);
       expect(playthrough.actions).toEqual([]);
       expect(playthrough.issueCustomizationArgs).toEqual({});
     });
   });
 
-  describe('recording exploration playthroughs', function() {
-    beforeEach(function() {
+  describe('recording exploration playthroughs', () => {
+    beforeEach(() => {
       this.expId = 'expId';
       this.expVersion = 1;
       this.playthroughRecordingProbability = 1.0;
 
-      this.PlaythroughService.initSession(
+      playthroughService.initSession(
         this.expId, this.expVersion, this.playthroughRecordingProbability);
-      spyOn(this.ExplorationFeaturesService, 'isPlaythroughRecordingEnabled')
+      spyOn(explorationFeaturesService, 'isPlaythroughRecordingEnabled')
         .and.returnValue(true);
     });
 
-    describe('.recordExplorationStartAction()', function() {
-      it('adds a learner action object to the actions array', function() {
-        this.PlaythroughService.recordExplorationStartAction('initStateName1');
+    describe('.recordExplorationStartAction()', () => {
+      it('adds a learner action object to the actions array', () => {
+        playthroughService.recordExplorationStartAction('initStateName1');
 
-        var playthrough = this.PlaythroughService.getPlaythrough();
+        var playthrough = playthroughService.getPlaythrough();
         expect(playthrough.actions).toEqual([
-          this.LearnerActionObjectFactory.createNew('ExplorationStart', {
+          learnerActionObjectFactory.createNew('ExplorationStart', {
             state_name: {value: 'initStateName1'},
-          }),
+          }, 1),
```

Comment:
Please use a constant name (`ACTION_SCHEMA_VERSION`) instead to make this more obvious.

Reply:
Done.